src/modules/peek/Peek.UI/MainWindowViewModel.cs

```diff
@@ -23,6 +25,16 @@ public partial class MainWindowViewModel : ObservableObject
         [ObservableProperty]
         private IFileSystemItem? _currentItem;
 
+        partial void OnCurrentItemChanged(IFileSystemItem? value)
```

Comment:
Nice! Where does this method come from? 🤔

Reply:
It comes from the powerful source generators of the `CommunityToolkit.Mvvm` package: https://learn.microsoft.com/dotnet/communitytoolkit/mvvm/generators/observableproperty#running-code-upon-changes